### PR TITLE
Implement backend of comments system

### PR DESCRIPTION
**Pull request by mdevlamynck**
Backend for #16.

It might be interesting to find a way to do functional tests to test server.rs.

It might also be interesting to abstract the pagination logic so we can reuse it instead of duplicating it in every paginated route.

### CODE REVIEW

File: migrations/20171007182038_create_comments/up.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE comments (
+  id SERIAL PRIMARY KEY,
+  message VARCHAR NOT NULL,
+  link_id INTEGER NOT NULL REFERENCES links (id),
+  author_id INTEGER NOT NULL REFERENCES users (id),
+  time INT NOT NULL,
```
Comment (huytd) [inline migrations/20171007182038_create_comments/up.sql]: Seems like the last comma caused error during migration process

File: src/bin/server.rs [hunk 2]
```diff
@@ -138,6 +139,47 @@ fn feed(page: i64) -> Json<Value> {
     }))
 }
 
+#[get("/link/<link>/comments")]
```
Comment (huytd) [inline src/bin/server.rs]: Just about the naming, I'd like to call it `link_id` instead of `link`

File: src/bin/server.rs [hunk 3]
```diff
@@ -15,16 +14,18 @@ use self::codedaily_backend::middleware::Auth;
...
 use rocket_contrib::{Json, Value};
 use std::io;
+use std::cmp::max;
 use std::path::{Path, PathBuf};
 use rocket::response::NamedFile;
 use crypto::digest::Digest;
 use crypto::sha2::Sha256;
 
 const LINKS_PER_PAGE: i64 = 30;
+const COMMENTS_PER_PAGE: u32 = 30;
```
Comment (huytd) [inline src/bin/server.rs]: Do you think we should move this constant (and the `LINKS_PER_PAGE` as well) into a configurable file, so we can change it  anytime?

File: src/bin/server.rs (hunk 2 above)
Comment (mdevlamynck) [reply src/bin/server.rs]: You're right.

I'm used to write controllers taking not the id of a model but the model itself (thing we may be able to do with a rocket guard?). The cool thing about that is you don't need to query the model and that if the id doesn't match any row in database it automatically returns a 404.

File: src/bin/server.rs (hunk 3 above)
Comment (mdevlamynck) [reply src/bin/server.rs]: Might be interesting. I'm not sure these values would change all that much though (changing enough that recompiling would be a bother). Usually this is more about fine tuning for performance and once it's settled it doesn't change.

That said we could just do it for the sake of doing it :wink:

File: src/bin/server.rs (hunk 2 above)
Comment (mdevlamynck) [reply src/bin/server.rs]: Ah, `link_id` conflicts with the name of the field imported into scope by `use codedaily_backend::schema::comments::dsl::*;`. So we have to either rename the parameter or the alias the imported field. Which do you prefer?

**Pull request merged**